### ISSUE TITLE
[a11y BEST PRACTICE] - Back to top component is not materially honest and should be reviewed

### ISSUE THREAD

**Issue by GnatalieH** [department-of-veterans-affairs/vets-design-system-documentation#899]
## Issue Description

We need to conduct a full accessibility review of the Back to top component. Months ago Josh Kim and Angela Fowler identified the Back to top component as not materially honest because it should be a link but instead is styled as a button. There may be an issue with the element living outside the DOM, as well. 

The component can be viewed in Storybook [here](https://design.va.gov/storybook/?path=/docs/components-va-back-to-top--default).

## Tasks
- [ ] Full accessibility review of the component is completed
- [ ] Add descriptions of any issues found to this ticket
- [ ] If any problems are found, add screen recordings and/or screenshots to comments on this ticket that show the problem.
- [ ] If necessary, discuss any accessibility issues found with the engineer who developed the component
- [ ] Re-review any fixes the developer makes to the component

## Acceptance Criteria
- [ ] Component has been reviewed for accessibility issues
- [ ] Any accessibility issues have been addressed

**Comment by SarahKay8** [comment]
Hi all. I wouldn't consider this launch blocking because there are other alternatives for users to navigate to the top of the web page. Also, because it's not labelled, this would be a 508 Defect 2..

**Comment by GnatalieH** [comment]
Thanks @SarahKay8, I went ahead and labeled it. I wasn't sure when I created the ticket what level defect it would be.

**Comment by SarahKay8** [comment]
With a label we can drop it to a 3 @caw310

**Comment by caw310** [comment]
Thanks @SarahKay8

**Comment by joshkimux** [comment]
@SarahKay8 , @GnatalieH and @caw310 , @artsymartha68 recently conducted an axe scan on this component on mobile only to uncover that it's not labelled (at all) due to the text label (nested in a span) being set to display none. As a result, on mobile screen readers it will be read out as "button"

https://user-images.githubusercontent.com/14154792/183979180-065eae3d-7009-4e31-a026-ff3b82cd5d15.mov

We should prioritize this as a `508-defect-1` as this does not meet Section 508 compliance. Simply keeping the label (not using display none) from the mobile view will fix this.

<img width="320" alt="Screen Shot 2022-08-10 at 1 37 30 PM" src="https://user-images.githubusercontent.com/14154792/183979483-c514ca09-cb1d-486a-b32b-e8a09f2905db.png">

I also still recommend we look into converting this into a link and not a button.

**Comment by bkjohnson** [comment]
I don't understand why there's a desire for this to be a link when it is [doing something](https://github.com/department-of-veterans-affairs/component-library/blob/ff6baa66f08b809d2bf3c4276c3da4f034b3968e/packages/web-components/src/components/va-back-to-top/va-back-to-top.tsx#L49-L63) - focusing the `<h1>` element and scrolling to the top of the page.

**Comment by joshkimux** [comment]
@bkjohnson I can break this down! It's a _little_ bit of an exception, but understandably confusing/awkward.

The main _purpose_ of the back to top component is to _go_ to a location on the page. Because the user is navigating to a place (scrolling) and nothing is being added (like a modal), removed (like deleting a card from the list loop), or state changed (like going into an edit mode), we can think of this as more of a link than a button.

You can think of this like the skip link or the on this page link. Both of these are links that send focus to a heading and scroll to a different part of the page.

And on that note... the sending focus part is a bit tricky though! Ideally, we wouldn't need to send focus there at all, but we need to do that because of some browser/assistive technology combinations. So it's kind of more like a work around 😓

**Comment by caw310** [comment]
Fixing this issue will require assistance from a11y specialist so this may take multiple sprints to close.

**Comment by caw310** [comment]
Will carry over as we wait for a11y feedback.

**Comment by humancompanion-usds** [comment]
In my mind this just requires a redesign where this is a link and not a button. I can work on this with @jamigibbs. I'll redesign this component this afternoon.

**Comment by jamigibbs** [comment]
@humancompanion-usds I think that's the only critical feedback we've received from the audit. They have provide several link examples for us to model ours after: https://github.com/department-of-veterans-affairs/va.gov-team/issues/60812#issuecomment-1610054544

**Comment by humancompanion-usds** [comment]
@jamigibbs - Here is my redesign following the Scottish Design System: 
https://sketch.com/s/73212626-acc4-47e6-acd8-d4e2f5a98372

Positioning stays the same. Really just the colors change and I got rid of the rounded corners (except for the focus state which is rounded). Oh and, obviously, made it a link and not a button. Let me know if you have questions.

**Comment by humancompanion-usds** [comment]
I owe Jami a hover, focus, and active state. Working on that today.
